docs/jdbc.md

```diff
@@ -0,0 +1,495 @@
+config.setJdbcUrl(jdbcURL);
+config.setDataSourceProperties(connProps);
+config.setConnectionTimeout(10000); // 10s
+
+HikariDataSource connectionPool = new HikariDataSource(config);
+```
+
+##### SQL Server
+
+Not Supported.
+
+
+#### Delegated Service Account Impersonation
+
+In addition, the `cloudSqlDelegates` property controls impersonation delegation.
+The value is a comma-separated list of service accounts containing chained
+list of delegates required to grant the final access_token. If set,
+the sequence of identities must have "Service Account Token Creator" capability
+granted to the preceding identity. For example, if set to
+`"serviceAccountB,serviceAccountC"`, the application default credentials must
+have the Token Creator role on serviceAccountB. serviceAccountB must have
+the Token Creator on serviceAccountC. Finally, C must have Token Creator on
+`cloudSqlTargetPrincipal`. If unset, the application default credential principal
+must "Service Account Token Creator" capability granted that role on the
+cloudSqlTargetPrincipal service account.
+
+
+```java
+connProps.setProperty("cloudSqlTargetPrincipal", "TARGET_SERVICE_ACCOUNT");
+connProps.setProperty("cloudSqlDelegates", "SERVICE_ACCOUNT_1,SERVICE_ACCOUNT_2");
+```
+
+In this example, the environment's application default principal impersonates
+SERVICE_ACCOUNT_1 which impersonates SERVICE_ACCOUNT_2 which then
+impersonates the TARGET_SERVICE_ACCOUNT.
+
+
+
+### Connection via Unix Sockets
+
+To connect using a Unix domain socket (such as the one created by the Cloud SQL
+proxy), you can use the `unixSocketPath` property to specify a path to a local
+file instead of connecting directly over TCP.
+
+##### Mysql
+
+```
+jdbc:mysql:///<DATABASE_NAME>?unixSocketPath=</PATH/TO/UNIX/SOCKET>&cloudSqlInstance=<INSTANCE_CONNECTION_NAME>&socketFactory=com.google.cloud.sql.mysql.SocketFactory&user=<MYSQL_USER_NAME>&password=<MYSQL_USER_PASSWORD>
+```
+
+##### Maria DB
+
+Not Supported.
+
+##### Postgres
+
+```
+jdbc:postgresql:///<DATABASE_NAME>?unixSocketPath=</PATH/TO/UNIX/SOCKET>&cloudSqlInstance=<INSTANCE_CONNECTION_NAME>&socketFactory=com.google.cloud.sql.postgres.SocketFactory&user=<POSTGRESQL_USER_NAME>&password=<POSTGRESQL_USER_PASSWORD>
+```
+
+##### SQL Server
+
+Not Supported.
+
+
+## Examples
+
+Examples for using the Cloud SQL JDBC Connector for SQL Server can be found by looking at the integration tests in this repository.
+* [MySQL connector-j-8](../jdbc/mysql-j-8/src/test/java/com/google/cloud/sql/mysql/JdbcMysqlJ8IntegrationTests.java)
+* [Maria DB Usage example](../jdbc/mariadb/src/test/java/com/google/cloud/sql/mariadb/JdbcMariaDBIntegrationTests.java)
+* [Postgres Usage example](../jdbc/postgres/src/test/java/com/google/cloud/sql/postgres/JdbcPostgresIntegrationTests.java)
+* [SQL Server Usage example](../jdbc/sqlserver/src/test/java/com/google/cloud/sql/sqlserver/JdbcSqlServerIntegrationTests.java)
+
+## Reference Documentation
+
+#### Mysql & Maria DB
```

Comment:
this subheading should be type 2, i.e. `##`

Same for the other subheadings below.

Reply:
Fixed.